packages/zendesk-support-adapter/src/client/client.ts

```
@@ -71,4 +89,56 @@ export default class ZendeskClient extends clientUtils.AdapterHTTPClient<
       throw e
     }
   }
+
+  public async ensureLoggedIn(): Promise<void> {
+    await super.ensureLoggedIn()
+    if (!this.isResourcesApiLoggedIn && this.resourcesConn) {
+      if (this.resourceasLoginPromise === undefined) {
+        this.resourceasLoginPromise = this.resourcesConn.login(this.credentials)
+      }
+      const resourcesClient = await this.resourceasLoginPromise
+      if (this.resourcesClient === undefined) {
+        this.resourcesClient = resourcesClient
+        this.isResourcesApiLoggedIn = true
+      }
+    }
+  }
+
+  @throttle<clientUtils.ClientRateLimitConfig>({ bucketName: 'get', keys: ['url', 'queryParams'] })
+  @logDecorator(['url', 'queryParams'])
+  @requiresLogin()
+  public async getResource(
+    args: clientUtils.ClientBaseParams,
+  ): Promise<clientUtils.Response<clientUtils.ResponseValue | clientUtils.ResponseValue[]>> {
+    if (this.resourcesClient === undefined) {
+      // initialized by requiresLogin (through ensureLoggedIn in this case)
+      throw new Error(`uninitialized ${this.clientName} client`)
+    }
+    try {
+      const { url, queryParams, headers, responseType } = args
+      const requestConfig = [queryParams, headers, responseType].some(values.isDefined)
+        ? {
+          params: queryParams,
+          headers,
+          responseType,
+        }
+        : undefined
+      const { data, status } = await this.resourcesClient.get(url, requestConfig)
+      log.debug('Received response for %s (%s) with status %d', url, safeJsonStringify({ url, queryParams }), status)
+      log.trace('Full HTTP response for %s: %s', url, safeJsonStringify({
+        url, queryParams, response: data,
+      }))
+      return {
+        data,
+        status,
+      }
+    } catch (e) {
+      const status = e.response?.status
+      if (status === 404) {
+        log.warn('Suppressing %d error %o', status, e)
```

is it really valid to suppress the error in our case? doesn't it mean the resource is unavailable? @noamhammer wdyt?